backend/s3_backend.go

```diff
@@ -156,7 +163,6 @@ func (s *S3Backend) ListFiles(db, version string) ([]string, error) {
 
 	log.Printf("call_site=s3.ListFiles sequins_db=%q sequins_db_version=%q dataset_size=%d file_count=%d", db, version, datasetSize, numFiles)
```

Comment:
I'm a bit back and forth on whether when logging, we should log the original version or the target version 😕

Reply:
Log that we are using a symlink, db version should be based from current s3 path where sym link file is, and the symlinked_version would be the OG version.